raiden/tests/utils/mediation_fees.py

```diff
@@ -109,4 +119,63 @@ def get_initial_payment_for_final_target_amount(
     except UndefinedMediationFee:
         return None
 
-    return PaymentWithFeeAmount(total), fees
+    return FeesCalculation(total_amount=PaymentWithFeeAmount(total), mediators_cut=fees)
+
+
+class PaymentAmountCalculation(NamedTuple):
+    """
+    Represents the result of get_amounts_to_drain_channel_with_fees
+
+    Differ from FeesCalculation in the fact that this returns the amount to send
+    without any fees or fee estimates.
+    """
+
+    amount_to_send: PaymentAmount
+    mediators_cut: List[FeeAmount]
+    amount_with_fees: PaymentWithFeeAmount
+
+
+def get_amount_for_sending_before_fees(
```

Comment:
Note to reviewer: This is a second utility function building on top of `get_initial_payment_for_final_target_amount`. 

While `get_initial_payment_for_final_target_amount` takes the target amount you want to finally reach the target as input, this function here `get_amount_for_sending_before_fees` takes the amount we want to leave the initiator. Essentially how much we need the initiator's balance to be reduced.

This is useful for example in cases where we want to see what should the "send_value" be to spend the entire deposit.

The function returns the `amount_to_send`, which is what should go as the send transfer amount, a list of all the mediators fees as returned by `get_initial_payment_for_final_target_amount` and the total `amount_with_fees`.

Important note:

`amount_with_fees` can also be bigger than `amount_to_send + sum(mediators_cut)`. That is since we also have the "internal routing" fee estimation which adds a percentage over the entire transfer amount and that can end up being bigger than what the mediators would withhold.

Reply:
As far as i can see, this function returns the amount calculation with and without fees. This function should be renamed to reflect that. I don't think the name part `before_fees` is correct in this sense.

Reply:
Also, I think this function should be outside of the tests package. This is because at some point, we plan to expose an API endpoint to return the "amount for sending with fees" to the web UI so that the user knows what is the cost of a transfer.

Reply:
> This function should be renamed to reflect that

It returns all data both before and after fees. The name reflects the difference to the one that paul has commited in the same file.

> Also, I think this function should be outside of the tests package.

Let's leave that for later if at all. This is an open question as to how we will also handle functions like this that originate from the PFS. Ideally we would have either a library or a monorepo.